core/coreobjects/include/coreobjects/property_object_internal.h

```diff
@@ -37,6 +37,7 @@ DECLARE_OPENDAQ_INTERFACE(IPropertyObjectInternal, IBaseObject)
     virtual ErrCode INTERFACE_FUNC clone(IPropertyObject** cloned) = 0;
     virtual ErrCode INTERFACE_FUNC setPath(IString* path) = 0;
     virtual ErrCode INTERFACE_FUNC isUpdating(Bool* updating) = 0;
+    virtual ErrCode INTERFACE_FUNC hasUserReadAccess(IBaseObject* userContext, Bool * hasAccessOut) = 0;
```

Comment:
The context parameter can probably be IUser?

Reply:
In most cases we are passing serializer.getUser() to this method. It returns IBaseObject, since we did not want ISerializer to know about IUser. Therefore I wanted to wrap this type checking inside hasUserReadAccess() method.